tests/_test_msui/test_msui.py

```diff
@@ -328,3 +328,225 @@ def test_flight_track_io(self, mockload, mocksave, mockq, mocki, mockw):
         assert self.window.listFlightTracks.count() == 2
         assert os.path.exists(self.save_ftml)
         os.remove(self.save_ftml)
+
+class Test_MSUIMainWindow:
+    def setUp(self):
+        # Create a QApplication instance for Qt tests
+        self.app = QtWidgets.QApplication([])
+
+        # Create temporary directory for JSON files
+        self.temp_dir = tempfile.mkdtemp()
+
+        # Mock config_loader to return temporary directory
+        self.config_patcher = patch("mslib.msui.msui_mainwindow.config_loader")
+        self.mock_config = self.config_patcher.start()
+        self.mock_config.side_effect = lambda dataset=None, default=False: {
+            "data_dir": self.temp_dir,
+            "new_flighttrack_template": [("START", 0, 0), ("END", 10, 10)],
+            "new_flighttrack_flightlevel": 350,
+            "restore_views": True,
+            "filepicker_default": "qt"
+        }.get(dataset, {})
+
+        # Mock view_restoration.save_view_settings
+        self.save_view_settings_patcher = patch("mslib.msui.msui_mainwindow.view_restoration.save_view_settings")
+        self.mock_save_view_settings = self.save_view_settings_patcher.start()
+
+        # Mock view_restoration.restore_view_settings
+        self.restore_view_settings_patcher = patch("mslib.msui.msui_mainwindow.view_restoration.restore_view_settings")
+        self.mock_restore_view_settings = self.restore_view_settings_patcher.start()
+        self.mock_restore_view_settings.side_effect = lambda name: {
+            "global": {"flight_track_name": name, "waypoints": [{"location": "START", "lat": 0, "lon": 0, "flightlevel": 350}]},
+            "views": [{"view_type": "topview", "view_id": f"view_topview_{name}", "settings": "topview_settings"}]
+        }
```

Comment:
This is mocking the thing under test, which defeats the purpose of testing (i.e. it is no longer testing application code). As a rule of thumb, mocking should be used as little as possible and should only ever be necessary for external code outside of your control, IMO.